luigi_pipeline/seqr_loading_optimized.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import os
+import re
+
+import luigi
+import hail as hl
+
+from hail_scripts.v02.utils.elasticsearch_client import ElasticsearchClient
+from lib.hail_tasks import HailMatrixTableTask, HailElasticSearchTask, GCSorLocalTarget, MatrixTableSampleSetError
+from lib.model.seqr_mt_schema import SeqrSchema, SeqrVariantSchema, SeqrGenotypesSchema
+import seqr_loading
+# from gnomad_hail.utils import vep_or_lookup_vep
+
+logger = logging.getLogger(__name__)
+
+
+class SeqrValidationError(Exception):
+    pass
+
+class SeqrVCFToVariantHTTask(seqr_loading.SeqrVCFToMTTask):
+    """
+    Loads all annotations for the variants of a VCF into a Hail Table (parent class of MT is a misnomer).
+    """
+
+    def run(self):
+        mt = self.import_vcf()
+        mt = hl.split_multi_hts(mt)
+        if self.validate:
+            self.validate_mt(mt, self.genome_version, self.sample_type)
+        mt = HailMatrixTableTask.run_vep(mt, self.genome_version, self.vep_runner)
+        # We're now adding ref data.
+        ref_data = hl.read_table(self.reference_ht_path)
+        clinvar = hl.read_table(self.clinvar_ht_path)
+        hgmd = hl.read_table(self.hgmd_ht_path)
+
+        ht = SeqrVariantSchema(mt, ref_data=ref_data, clinvar_data=clinvar, hgmd_data=hgmd).annotate_all(
+            overwrite=True).select_annotated_mt().rows()
+
+        ht.describe()
+
+        ht.write(self.output().path, stage_locally=True)
+        hl.copy_log(self.output().path + '-logs.log')
+
+class SeqrVCFToGenotypesMTTask(HailMatrixTableTask):
+    remap_path = luigi.OptionalParameter(default=None,
+                                         description="Path to a tsv file with two columns: s and seqr_id.")
+    subset_path = luigi.OptionalParameter(default=None,
+                                          description="Path to a tsv file with one column of sample IDs: s.")
+    def run(self):
+        mt = self.import_vcf()
```

Comment:
Should we run this off the MT from the SeqrVCFToVariantHTTask instead of importing and splitting the vcf again?